src/lib/sectionList/sectionListViewsConfig.ts

```diff
@@ -0,0 +1,192 @@
+import i18n from '@dhis2/d2-i18n'
+import { flow, get } from 'lodash/fp'
+import { getTranslatedProperty } from '../constants/translatedModelProperties'
+import { uniqueBy } from '../utils'
+import type { ConfigurableFilterKey } from './filters'
+
+export interface ModelPropertyDescriptor {
+    label: string
+    path: string
+}
+
+export interface FilterDescriptor {
+    label: string
+    filterKey: ConfigurableFilterKey
+}
+
+type Descriptor =
+    | (ModelPropertyDescriptor & Partial<FilterDescriptor>)
+    | (FilterDescriptor & Partial<ModelPropertyDescriptor>)
+
+/* Configs can either define the label and filterKey, or a string
+If config is a string, getTranslatedProperty will be used to get the label. */
+type FilterConfig = ConfigurableFilterKey | FilterDescriptor
+type ModelPropertyConfig = string | ModelPropertyDescriptor
+
+interface UntranslatedViewConfigPart<TEntry> {
+    available: ReadonlyArray<TEntry>
+    overrideDefaultAvailable?: boolean
+    default: ReadonlyArray<TEntry>
+}
+
+interface UntranslatedViewConfig {
+    columns: UntranslatedViewConfigPart<ModelPropertyConfig>
+    filters: UntranslatedViewConfigPart<FilterConfig>
+}
+
+interface ViewConfigPart<TEntry> {
+    available: ReadonlyArray<TEntry>
+    default: ReadonlyArray<TEntry>
+}
+
+interface ViewConfig {
+    columns: ViewConfigPart<ModelPropertyDescriptor>
+    filters: ViewConfigPart<FilterDescriptor>
+}
+
+interface SectionListView {
+    [key: string]: ViewConfig
+}
+
+const DESCRIPTORS = {
+    publicAccess: {
+        path: 'sharing.public',
+        label: i18n.t('Public access'),
+        filterKey: 'publicAccess',
+    },
+} satisfies Record<string, Descriptor>
+
+const toModelPropertyDescriptor = (
+    propertyConfig: ModelPropertyConfig
+): ModelPropertyDescriptor => {
+    if (typeof propertyConfig === 'string') {
+        return {
+            label: getTranslatedProperty(propertyConfig),
+            path: propertyConfig,
+        }
+    }
+    return propertyConfig
+}
+
+export const toFilterDescriptor = (
+    propertyConfig: FilterConfig
+): FilterDescriptor => {
+    if (typeof propertyConfig === 'string') {
+        return {
+            label: getTranslatedProperty(propertyConfig),
+            filterKey: propertyConfig,
+        }
+    }
+    return propertyConfig
+}
+
+function mergeConfigs<TEntry>(...configs: ReadonlyArray<TEntry>[]) {
+    return uniqueBy(configs.flat(), get('path'))
```

Comment:
This wouldn't work because filters to not have a path, so you would need to either pass that in or a different function